### DISCUSSION TITLE
Exposure time finder calibration doesn't have UI for multi exposure

### DISCUSSION THREAD

**Discussion by echo-bravo-yahoo** [sn4k3/UVtools#890] [Q&A]
### System

```Text
UVtools v4.3.2 X64
Operative system: Microsoft Windows 10.0.19045 X64
Processor: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
Processor cores: 12
Memory RAM: 21.00 / 31.94 GB
Runtime: win10-x64
Framework: .NET 6.0.29
AvaloniaUI: 11.0.10
OpenCV: 4.8.1

Sreens, resolution, working area, usable area:
1: 1920 x 1080 @ 100% (Primary) (On this)
    WA: 1920 x 1040    UA: 1920 x 1040
2: 1080 x 1920 @ 100%
    WA: 1080 x 1880    UA: 1080 x 1880

Path:       C:\Program Files\UVtools\
Executable: C:\Program Files\UVtools\UVtools.exe
Loaded file: UVtools_demo_file.sl1s [Version: 0] [Class: SL1File]
```

### Printer and Slicer

- Printer: N/A
- Slicer: N/A

### Description of the bug

The UI for exposure calibration test references a multi-exposure setting, but doesn't have UI to configure it. Multiple brightness appears to only work by anti-aliasing.

![image](https://github.com/sn4k3/UVtools/assets/759892/5373322b-15d6-47d7-b2a7-0083bd6cdebc)

### How to reproduce

Open UVtools with a completely default install/no configuration. Go to Calibration  > Exposure time finder. Attempt to find settings for "Multiple exposure times per layer."

**Comment by echo-bravo-yahoo** [comment]
After importing sliced .goo file, UVTools now has the fields I expected in exposure time finder. I looked for ways to configure the printer I'm using in UVTools and couldn't find any - would a "Import printer settings from slicedfile" option in the File menu make sense? Alternatively, rendering the "Multiple exposures" section in the "Exposure time finder" dialog with help text ("your current printer doesn't support ... please import a sliced file ...") could help.

**Comment by echo-bravo-yahoo** [comment]
Now the "Exposure time finder" dialog does not actually overwrite the file I imported to be able to see the settings.

**Comment by echo-bravo-yahoo** [comment]
Ok... This workflow worked for me:
1. Import a sliced .goo file so UVTools knows printer dimensions & believes multi exposure times will work.
2. Generate a _non-patterned_ exposure test. UVTools remembers the printer settings.
3. Generate a _patterned_ exposure test. Doing this directly fails (no test is generated, and I'm still viewing the imported, sliced .goo from earlier).

**Reply by sn4k3** [reply]
Hidden or grey-out features means not supported by your printer and/or file format.

1. You always need to open a file you printed before or for your printer to generate tests.
2. 
3. The pattern will only work if there's enough XY space to pattern the model.

**Reply by echo-bravo-yahoo** [reply]
There's still at least 1 code bug here - doing step #1, then step #3, without step #2 in the middle, fails silently. The modal closes, but no progress bar for exposure test creation appears and I remain looking at the imported file in the viewer. The XY space is the same since I've already imported the sliced .goo file. If there are any log files I can provide of this workflow, let me know where to look.

I also think there's a usability bug here as well - hiding the multiple exposures panel means I don't have any feedback on why it's unusable, whereas including a (disabled) panel with helptext would guide me to figuring it out faster.

**Reply by sn4k3** [reply]
If you talking about "Pattern load model" its normally to fail. But the "Multiple exposures" should always generate the tests.
If you believe there is a bug, share the file and write down the exact steps to reproduce.

> I also think there's a usability bug here as well - hiding the multiple exposures panel means I don't have any feedback on why it's unusable, whereas including a (disabled) panel with helptext would guide me to figuring it out faster.

It's true, I can review that in next patch.
